src/librustc_passes/diagnostics.rs

```diff
@@ -228,4 +228,5 @@ pub impl Foo for Bar {
 register_diagnostics! {
     E0472, // asm! is unsupported on this target
     E0561, // patterns aren't allowed in function pointer types
+    E0571, // `break` with an argument in a `while` loop
```

Comment:
What about `break` in `for` loops? Can we investigate desugaring `while` loops into regular `loop` instead of adding this special case?